New packages: Hyperbolic_triangulation_2 / Periodic_4_hyperbolic_triangulation_2

imiordanov opened the pull request:
## Summary of Changes

Creating a pull request for this branch as indicated by Mael. This is part of the review process for two new packages submitted for integration in CGAL: Hyperbolic_triangulation_2 and Periodic_4_hyperbolic_triangulation_2. Both packages are hosted on the same branch (because one depends on the other), this is why I am creating a single PR. 

## Release Management

* Affected package(s): two new packages: Hyperbolic_triangulation_2 and Periodic_4_hyperbolic_triangulation_2
* Feature pages:
   * [Hyperbolic_triangulation_2](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Hyperbolic_triangulation_2) 
   * [Periodic_4_hyperbolic_triangulation_2](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Periodic_4_hyperbolic_triangulation_2)

TODO:
- [x] Modify CHANGES.md
- [x] Filter branch, current weight is 4498kB (please notify @sloriot when it is the last item to be done)

TODO (web):
- [ ] Post this: https://gist.github.com/MaelRL/b161f890b318aa4b05853d0754de7ddf when the PR is integrated

sloriot commented:
Will this PR be the official one after the acceptance of the package? If the PR is only here for review purpose, note that an alternative solution is to create a PR in https://github.com/imiordanov/cgal

imiordanov commented:
Hi @sloriot ,

Please excuse my ignorance, but I am not sure that I understand what you mean -- should I push onto the `master` branch of https://github.com/imiordanov/cgal and create a PR against that one? Or is it something else?

Regarding your question, I suppose that this PR could also be the official one after the acceptance of the package. Of course, I am willing to follow your indications in order to have a smoother and cleaner integration, and also to make your (and my) life easier :)

sloriot commented:
I mean if this PR is not temporary and only meant for review purpose only it is fine. Otherwise, a PR in a `mine` remote might be better (like for example what we did [here](https://github.com/CUBERLEON/cgal/pull/1)). AFAIU, everything seems good in your case.

imiordanov commented:
OK, I think I see what you mean. Thank you for your comment, I will keep that in mind!

lrineau commented:
@maxGimeno:
Could you please have a look at what is required in this PR to make Travis happy?

@imiordanov @MaelRL @gdamiand @pougetma @MoniqueTeillaud @afabri:
It seems of lot of work has been done since the feature freeze, and the review of the API. What is the status of the review of the user manual? The wiki pages seem to indicate that there is an unreviewed submission of the user manual, in both features.

MaelRL commented:
I'm looking at the user manuals (and probably the code) later today.

maxGimeno commented:
It says "travis.yml has changed", I guess @imiordanov added the packages to the list in travis.yml instead of calling .travis/genereate_travis, which would have done it itself.
The fix is to do that, meaning cd to cgal_root and call 

bash .travis/generate_travis

But before that it would be safer to fix the diff that is way too big for what it should be, and contains unwanted modifications to the travis files.

imiordanov commented:
Hello all,

As far as I can remember, I don't think I've edited the travis.yml file... 
Is it possible that I had a very old version, and somehow when I was updating my branch to the latest version of `master` it replaced the file that should be there?

Is it safe to take the travis file from `master`, run the script indicated by @maxGimeno, and then commit the generated travis file?

maxGimeno commented:
@imiordanov travis.yml will be erased by the script, no need to take it from master. It would need the files template.txt and generate_travis.sh, and would be what I normally recommend, but I really think there is something wrong with this github state. As @lrineau says, there shouldn't be 1728 modified files, and I'm rather sure that  travis.yml is among the ones that shouldn't have changed. Fixxing that should also fix the travis problems. Once it is done you will still have to run the script.

lrineau commented:
I have changed the "base" of the PR twice, to force Github recompute the diff. And now it looks like it should be.

lrineau commented:
But, anyway, @maxGimeno, the .travis.yml and the template *should be* changed, because this pull-request adds two new packages.

maxGimeno commented:
the template doesn't need any change. And that's what I said, now tha tit is clean, the script needs to be called to update the list of packages.

lrineau commented on Documentation/doc/Documentation/dependencies:
```diff
@@ -97,3 +97,5 @@ Surface_mesh_shortest_path
 Set_movable_separability_2
 Classification
+Hyperbolic_triangulation_2
+Periodic_4_hyperbolic_triangulation_2
```
This line does not end with an end-of-line. That is incorrect, and some tools may complain, or ignore the last line completely.

maxGimeno commented:
Sorry I was wrong, the generate_travis.sh must be called from  CGAL_ROOT/.travis/

MaelRL commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_CK_traits_2.h:
```diff
@@ -0,0 +1,589 @@
+  Hyperbolic_Delaunay_triangulation_CK_traits_2() 
+  {}
+  
+  Hyperbolic_Delaunay_triangulation_CK_traits_2(const Hyperbolic_Delaunay_triangulation_CK_traits_2 & other)
```
I got rid of it and it seems fine.

MaelRL commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_CK_traits_2.h:
```diff
@@ -0,0 +1,589 @@
+  
+  Compare_x_2 
+    compare_x_2_object() const 
+  { return Compare_x_2();} 
```
See `Hyperbolic_Delaunay_triangulation_traits_2.h` to get an example of the reworked traits class.

imiordanov replied:
Done for CK as well.

MaelRL commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_CK_traits_2.h:
```diff
@@ -0,0 +1,589 @@
+    Hyperbolic_segment_2 operator()(const Hyperbolic_point_2& p, const Hyperbolic_point_2& q) const
+    {
+      Origin o;
+      if(Euclidean_collinear_2()(p, q, Hyperbolic_point_2(o))){
```
See Hyperbolic_Delaunay_triangulation_traits_2.h to get an example of the reworked traits class.

imiordanov replied:
Thank you for reworking the traits class! I've done the same for the CK traits, and also factorized the code (see below).

lrineau commented:
@MaelRL: Please tell us once the pull-request is ready to be tested.

@maxGimeno: This PR is about a (full) feature, that just got fully accepted. It is almost ready to be tested. Once @MaelRL tell us it is ready, you should try to test it as soon as possible, because that is a big one.

Cc: @afabri

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/examples/Periodic_4_hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+
+  	include( CGAL_CreateSingleSourceCGALProgram )
+
+  	include_directories (BEFORE "../../include" $HOME/include)
```
Same issue

imiordanov replied:
Cleaned.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,688 @@
+  result_type operator()(const Point& p0, const Point& p1,
+                         const Hyperbolic_translation& o0, const Hyperbolic_translation& o1) const
+  {
+    return Predicate()(pp(p0, o0), pp(p1, o1));
```
The adaptor should derive from the predicate or have a const& to it, so that it doesn't default construct the predicate every time it calls it. See periodic_3 traits and their adaptors.

imiordanov replied:
Indeed, I have modified the adaptor to mimic the adaptors for Periodic_3.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,688 @@
+  }; // end Construct_inexact_hyperbolic_circumcenter_2
+
+public:
+  Periodic_4_hyperbolic_Delaunay_triangulation_traits_2() {}
```
That constructor should take also be `(const Base& base = Base()) : Base(base)` etc.

imiordanov replied:
Done, as for HT2.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,688 @@
+
+  Orientation_2
+  orientation_2_object() const
+  { return Orientation_2(); }
```
This is unneeded, the base already defines this.

lrineau replied:
@MaelRL Do you agree with the answer? On the PR page, this thread is now the only open discussion.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,688 @@
+
+  Construct_hyperbolic_line_2
+  construct_hyperbolic_line_2_object() const
+  { return Construct_hyperbolic_line_2(); }
```
For all these predicates / constructions, you either don't need them, or should do the same as in Hyperbolic_Delaunay_triangulation_traits.

imiordanov replied:
Done, I have changed the traits class to follow Hyperbolic_Delaunay_triangulation_traits.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,688 @@
+      Circle_2 circle(center, sq_radius);
+      std::pair<Hyperbolic_point_2,Hyperbolic_point_2> inters = Construct_intersection_2()(circle,Circle_2(Hyperbolic_point_2(0,0),1));
+
+      if(Orientation_2()(circle.center(), inters.first, inters.second) == POSITIVE) {
```
see Hyperbolic_Delaunay_triangulation_traits

imiordanov replied:
Done as well.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,687 @@
+public:
+  Point dual(const Face_handle f, const Hyperbolic_translation& nboff = Hyperbolic_translation()) const
+  {
+    Point res = Construct_hyperbolic_circumcenter_2()(f->vertex(0)->point(),
```
Need to call the traits' (`geom_traits().construct...`) + same at other places

imiordanov replied:
Checked and done.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_triangulation_vertex_base_2.h:
```diff
@@ -0,0 +1,115 @@
+
+  Periodic_4_hyperbolic_triangulation_vertex_base_2(const Face_handle /*fh*/)
+    : Base(), _stored_translation(false)
+  {}
```
Is this constructor really needed?

imiordanov replied:
In fact, it was not, so I removed it.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/package_info/Periodic_4_hyperbolic_triangulation_2/description.txt:
```diff
@@ -0,0 +1,3 @@
+Package 2D periodic hyperbolic triangulation: provides Delaunay triangulations
+of the Bolza surface, represented as periodic triangulations in the Poincaré disk 
+model of the hyperbolic plane, with demos, examples, and tests.  
```
Both HT2 and P4HT2 are missing the long_descriptor file

imiordanov replied:
I've added this file to both packages.

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/test/Periodic_4_hyperbolic_triangulation_2/test_p4ht2_complex.cpp:
```diff
@@ -0,0 +1,57 @@
+#include <CGAL/CORE_Expr.h>
```
Filename looks strange with respect to what this tests.

imiordanov replied:
Modified filename to `test_p4ht2_exact_complex_numbers.cpp`

MaelRL commented on Periodic_4_hyperbolic_triangulation_2/test/Periodic_4_hyperbolic_triangulation_2/test_p4ht2_remove_dummy_points.cpp:
```diff
@@ -0,0 +1,94 @@
+#include <CGAL/Cartesian.h>
+
+#include <CGAL/point_generators_2.h>
+//#include <CGAL/Random_points_in_disc_2.h>
```
Can be cleaned from the multiple places it appears in.

imiordanov replied:
I've cleaned the code of the tests to remove unneeded comments.

imiordanov commented on Hyperbolic_triangulation_2/benchmark/Hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+
+  	include( CGAL_CreateSingleSourceCGALProgram )
+
+  	include_directories (BEFORE "../../include" $HOME/include)
```
I've removed the "include_directories" command from where it was not needed. For the time being I'm cleaning HT2, I will do the same for P4HT2.

imiordanov replied:
Cleaned from both packages.

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,956 @@
+  }
+  
+  // check if a star is bounded by finite faces
+  // TODO: rename this function name
```
I have implemented most of the missing functions. The only things that remain as \todo are the `operator==` and `operator!=` functions in both `HT2` and `P4HT2`, but I wonder if they are really needed... 

I have found these operators implemented for `Triangulation_3`, `Periodic_3_triangulation_3`, and `Periodic_2_triangulation_2`, so it might make sense to have them for `HT2` and `P4HT2`, but I want to make sure that they're needed before implementing them, as it will take me some time, from what I saw in the existing methods.

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_old_traits_2.h:
```diff
@@ -0,0 +1,468 @@
+// Copyright (c) 2010-2016   INRIA Sophia Antipolis, INRIA Nancy (France).
```
This is an old traits class written by Misha that we resuscitated to make some performance benchmarks. Perhaps I shouldn't have committed it...

imiordanov replied:
The file has been moved to another branch (INRIA/Periodic_2g_hyperbolic_triangulation_2-IIordanov)

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,956 @@
+  
+  void mark_star_faces(Vertex_handle v) const
+  {
+    // TODO: think of it
```
Also done! I think... :thinking:

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,956 @@
+  {
+    Is_Delaunay_hyperbolic del;
+    int idx;
+    bool flag = del(f->vertex(0)->point(),f->vertex(1)->point(),f->vertex(2)->point(),idx);
```
The functions have been implemented and used where appropriate.

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,956 @@
+    // hyperbolic line and ray
+    Hyperbolic_segment ray = this->geom_traits().construct_hyperbolic_bisector_2_object()(p,q,hyp_face->vertex(i)->point());
+    // TODO MT cut edge at dual point !!!
+    //    Segment ray = this->geom_traits().construct_ray_2_object()(dual(finite_face), line);
```
In fact, this has been already done. Cleaned the comments.

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_traits_2.h:
```diff
@@ -0,0 +1,746 @@
+// Copyright (c) 2016  INRIA Nancy - Grand Est (France).
```
I have factorized a few common things, but unfortunately, I couldn't factorize everything: the CK and the CORE traits use different intersections and it's a mess to resolve the dependencies, so I had to leave circumcenters and bisectors in the respective classes.

imiordanov commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_CK_traits_2.h:
```diff
@@ -0,0 +1,589 @@
+  typedef Construct_hyperbolic_segment_2 Construct_segment_2;
+  Construct_segment_2
+    construct_segment_2_object() const
+  { return Construct_hyperbolic_segment_2(); }
```
I see your point. I've changed the declaration, now `construct_segment_2_object()` returns a Euclidean segment. This misnaming was done just for the demo, but it has been taken care of.

lrineau commented:
I think we want this feature (and pull-request) into CGAL-4.14. But the code freeze is by the end of the week, and:
  - Travis reports compilation errors (maybe C++11?),
  - the pull-request has never been tested.

First you have to find out what is wrong with the the compilation errors on Travis, and fix them. And then we will test it in the CGAL `integration` branch.

And that would be nice to at least start testing it in `integration` before the end of the week.

imiordanov commented:
Hello @afabri, I have just added you as collaborator.
Thank you for the tip, I am just going through the list now! I will try to make fixes asap.

imiordanov commented on Hyperbolic_triangulation_2/demo/Hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -0,0 +1,41 @@
+
+# Instruct CMake to run moc automatically when needed.
+set(CMAKE_AUTOMOC ON) 
+set(CMAKE_AUTOUIC ON)
```
Will do, thank you @maxGimeno! 😃

MaelRL commented on Hyperbolic_triangulation_2/tests/Hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -0,0 +1,28 @@
+# This is the CMake script for compiling a CGAL application.
+
+
+project( Hyperbolic_triangulation_2_tests )
```
Must be `Tests` (with a capital `t`) to make automated testing happy.

imiordanov replied:
Thank you for editing this, @MaelRL !
Both HT2 and P4HT2 follow this rule now.

imiordanov replied:
...which I just committed and pushed... 😅

MaelRL commented:
HT2 seems good for Travis now (modulo testing of Tests, which isn't done currently), but P4HT2 has some errors (https://travis-ci.org/CGAL/cgal/jobs/476751096#L2704).

MaelRL commented on Hyperbolic_triangulation_2/include/CGAL/Hyperbolic_Delaunay_triangulation_2.h:
```diff
@@ -0,0 +1,956 @@
+      }
+      Hyperbolic_adjacent_vector_circulator(Vertex_handle v, Face_handle fh = Face_handle()): VBase(v, fh) {
+        _v = v;
+        if (fh = Face_handle()) 
```
This comment is the last one that needs a look, afaik.

imiordanov commented:
Hello @MaelRL and @lrineau, 
I've added tests for HT2 and cleaned the tests for P4HT2. Travis also looks happier now. 
What should I do for the test in the `integration` branch?

afabri commented:
> 
> 
> Hello @MaelRL and @lrineau,
> I've added tests for HT2 and cleaned the tests for P4HT2. Travis also looks happier now.
> What should I do for the test in the `integration` branch?

Follow the [instructions ](https://github.com/CGAL/cgal/wiki/Developing-Features-with-Git#integrating-the-feature-branch-and-adding-it-to-the-testsuite)in order to merge it into integration.

imiordanov commented:
> > Hello @MaelRL and @lrineau,
> > I've added tests for HT2 and cleaned the tests for P4HT2. Travis also looks happier now.
> > What should I do for the test in the `integration` branch?
> 
> Follow the [instructions ](https://github.com/CGAL/cgal/wiki/Developing-Features-with-Git#integrating-the-feature-branch-and-adding-it-to-the-testsuite)in order to merge it into integration.

Thank you @afabri, I'm on it!

imiordanov commented:
I have committed and pushed the new packages in `integration`. I guess that if Travis is happy with the latest changes, we can wait for the results of the test tomorrow. Please let me know if there is anything else left (I am still seeing that changes are requested, but not sure what...)

MaelRL commented:
The HT2 tests are in the wrong folder, it should be `HT2/test/HT2` and not `/test**s**/`.

afabri commented:
Examples that only print the usage information because they expect arguments should check argc and take defaults.
Let me know if there are warnings/errors with VC++ where you want me to have a look.

maxGimeno commented:
> The HT2 tests are in the wrong folder, it should be `HT2/test/HT2` and not `HT2/tests/HT2`.
> 
> Bit of an inconsistency, I guess...

The complete path is now `HT2/test/tests/HT2` instead of `HT2/test/HT2`, which must contain the CMakeLists. This is what bothers travs for now :)

imiordanov commented:
> > The HT2 tests are in the wrong folder, it should be `HT2/test/HT2` and not `HT2/tests/HT2`.
> > Bit of an inconsistency, I guess...
> 
> The complete path is now `HT2/test/tests/HT2` instead of `HT2/test/HT2`, which must contain the CMakeLists. This is what bothers travs for now :)

I tried to do fast, but I didn't do well...
Should be OK now, thank you! :)

maxGimeno commented:
I add this PR to integration now that travis is green. I'll post feedback here every day until the testsuite is all green, so you don't have to take care of integration anymore.

lrineau commented:
[**@imiordanov**](https://github.com/imiordanov) commented on [Aug 16, 2018, 
> TODO:
> - [ ] Modify CHANGES.md
> - [ ] Make some nice post for cgal.org :)
> - [ ] Filter branch, current weight is 14Mo

Hi Iordan. Please update this TODO list (by clicking on it in the first message).

- I think the `CHANGES.md` has been modified.
- As for the weight of the branch, I do not remember if the history was modified. @sloriot?

imiordanov commented:
> [**@imiordanov**](https://github.com/imiordanov) commented on [Aug 16, 2018,
> 
> > TODO:
> > 
> > * [ ]  Modify CHANGES.md
> > * [ ]  Make some nice post for cgal.org :)
> > * [ ]  Filter branch, current weight is 14Mo
> 
> Hi Iordan. Please update this TODO list (by clicking on it in the first message).
> 
> * I think the `CHANGES.md` has been modified.
> * As for the weight of the branch, I do not remember if the history was modified. @sloriot?

Hello @lrineau, 
Indeed, CHANGES.md has been modified, but I am not sure about the weight of the branch... also about the nice post about cgal.org -- what should I do there..?

MaelRL commented:
The branch is heavy and you can trim a lot of useless stuff from it (old versions of images, code that was placed at the wrong place, etc.). We will deal with it.

For the nice post, the idea is to make a short news post on cgal.org to present your new packages; something like 10 sentences and 3 pictures. You can look at https://www.cgal.org/2018/09/10/Periodic_Mesh_Generation/ for a (not necessarily perfect) example. The source is there: https://raw.githubusercontent.com/CGAL/cgal-web/master/_posts/2018-09-10-Periodic_Mesh_Generation.md?token=AE1s-gKhMEcmsvW6da7N6Zg_R2QrlJFTks5cP0UMwA%3D%3D. You can look into https://github.com/CGAL/cgal/wiki/Tools to see how to make a new post.

sloriot commented:
I'll have a look for filtering, don't spend time for now on that.

lrineau commented:
Actually, now the weight is 4449kB. Probably the history was already filtered.

afabri commented on Periodic_4_hyperbolic_triangulation_2/test/Periodic_4_hyperbolic_triangulation_2/test_p4ht2_matrix.cpp:
```diff
@@ -0,0 +1,47 @@
+#include <CGAL/CORE_Expr.h>
+#include <CGAL/Point_2.h>
```
Do not include Point_2.h and the like as including a kernel does this.

imiordanov replied:
Done.

maxGimeno commented:
Errors from the testsuite : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-92.shtml#Hyperbolic_triangulation_2

and 
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-92.shtml#Periodic_4_hyperbolic_triangulation_2

afabri commented on Periodic_4_hyperbolic_triangulation_2/test/Periodic_4_hyperbolic_triangulation_2/test_p4ht2_matrix.cpp:
```diff
@@ -0,0 +1,47 @@
+#include <vector>
+
+using namespace CGAL;
+using namespace std;
```
Do not use `using`

lrineau replied:
In a .cpp test file, why not?

afabri replied:
You certainly know better.  Would you also put a `using`  in CGAL examples?  I always had the feeling that it invites trouble for the disambiguation.

lrineau replied:
No, I would forbid the same code in an example, because they are part of the CGAL manual, and those using directives (on full namespaces) are generally a bad practice.

imiordanov replied:
I have removed the `using` statements.

afabri commented on Periodic_4_hyperbolic_triangulation_2/test/Periodic_4_hyperbolic_triangulation_2/test_p4ht2_matrix.cpp:
```diff
@@ -0,0 +1,47 @@
+  for(std::size_t i=0; i<gens.size(); ++i)
+    std::cout << "g[" << i << "] = " << gens[i] << std::endl;
+
+  CGAL_assertion(gens[0]*gens[4] == m);
```
Use `assert` in testsuite files.

imiordanov replied:
Done.

imiordanov commented:
> Errors from the testsuite : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-92.shtml#Hyperbolic_triangulation_2
> 
> and
> https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-92.shtml#Periodic_4_hyperbolic_triangulation_2

Thank you @maxGimeno for the links!

To be honest, I am a little confused... I see errors for things that I had pushed yesterday, so I don't know what I should do now: should I wait for the next testsuite results, should I commit/push again, or something else..?

Another things is that for the demo of HT2 the error is something I'm not quite sure how to fix: it complains about the file `qrc_Delaunay_triangulation_2.cpp`, which should be created automatically, cmake is supposed to automoc it. Am I doing something wrong?

If I understand correctly, the only thing that is not working (or, better, working randomly) is the test `ht2_test_copy.cpp`, which I will try to solve asap.

MaelRL commented:
There are some warnings in the doc (https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.14-Ic-92/logs_master/Hyperbolic_triangulation_2.log) as reported in #3585.

afabri commented:
> To be honest, I am a little confused... I see errors for things that I had pushed yesterday, so I don't know what I should do now: should I wait for the next testsuite results, should I commit/push again, or something else..?

Maxime merged into integration at about 4pm.  And you pushed fixes  later.

imiordanov commented:
> > To be honest, I am a little confused... I see errors for things that I had pushed yesterday, so I don't know what I should do now: should I wait for the next testsuite results, should I commit/push again, or something else..?
> 
> Maxime merged into integration at about 4pm. And you pushed fixes later.

I see, that explains everything.

With the last commit, I think there should be some warnings but no errors. Well, one error. The only error that I haven't solved yet and I'm not sure how to proceed is the one about the file `qrc_Delaunay_triangulation_2.cpp`. I'm guessing I need to do something in CMakeFiles.txt, but I am unable to reproduce the issue, so for now it remains unsolved.

maxGimeno commented:
@imiordanov There are platforms on docker that are unable to use MOC correctly because of a permission problem, you don't have to worry about it. If that is the error you are talking about, just forget about it. You can see all demos on those platforms are red.

afabri commented:
@imiordanov Have a look at `Filtered_kernel/include/CGAL/internal/Static_filters/Angle_3.h` to see what we have to do in the `#else` in order to please VC2013.

afabri commented:
@imiordanov @MoniqueTeillaud  @MaelRL   I am wondering what number type is really needed or recommended. I see inclusions of the Epic header but it seems not to be used.  I see `Cartesian<Expr>` elsewhere. Do you really need `Expr` ?     Would it be better with `Simple_cartesian`?  What about `Epec_with_sqrt`?

MoniqueTeillaud commented:
The computations are algebraically very heavy, with high degree. Expr is really needed.

afabri commented:
> 
> The computations are algebraically very heavy, with high degree. Expr is really needed.

Even then my other questions are still open.

imiordanov commented:
> @imiordanov @MoniqueTeillaud @MaelRL I am wondering what number type is really needed or recommended. I see inclusions of the Epic header but it seems not to be used. I see `Cartesian<Expr>` elsewhere. Do you really need `Expr` ? Would it be better with `Simple_cartesian`? What about `Epec_with_sqrt`?

Hello @afabri,
For HT2 we have two traits classes, the one is based on `Cartesian<CORE::Expr>` and the other one on `Circular_kernel_2<EPICK>` (plus filters). As @MoniqueTeillaud says, we need `Expr` for one of the traits because the computations are very demanding, and we chose `Cartesian` over `Simple_cartesian` for performance reasons (cf doc for `Simple_cartesian`: _In contrast to Cartesian, no reference counting is used internally. This eases debugging, but may slow down algorithms that copy objects intensively._)
I am not sure about `Epec_with_sqrt`... we need number types that support nested square roots (such as `sqrt(1+sqrt(2))`), would this type be suitable?

afabri commented:
@imiordanov did you check that Cartesian is really faster?   Epec_with_sqrt should be something with Expr inside.  There is no need to include Epic.

MoniqueTeillaud commented:
* If I remember correctly, we had tested Simple_Cartesian (which is not very surpising since there is no ref counting). In any case it cannot hurt to test again to make sure. 
* Honestly the manual is not explicit enough for me to guess whether Epec_with_sqrt is enough for our needs... 
@Andreas Can you elaborate on why you think that it would be better to use Simple_cartesian or Epec_with_sqrt? 
@Iordan Can you answer Andreas' concern: "I see inclusions of the Epic header but it seems not to be used."?

afabri commented:
>  @andreas Can you elaborate on why you think that it would be better to use Simple_cartesian or Epec_with_sqrt?

Because I don't know which of Cartesian/Simple_cartesian is better, but I am not sure that you have explored it. For Epec the reasoning is that we recommend the usage of Epic or Epec, so why should that be different for Epec_with_sqrt.

MoniqueTeillaud commented:
----- Le 14 Jan 19, à 10:16, Andreas Fabri <notifications@github.com> a écrit : 

> I don't know which of Cartesian/Simple_cartesian is better,
The manual says: " In contrast to [ https://doc.cgal.org/latest/Kernel_23/structCGAL_1_1Cartesian.html | Cartesian ] , no reference counting is used internally. This eases debugging, but may slow down algorithms that copy objects intensively. " 

> but I am not sure that you have explored it.
ok. As said earlier I think that we have tested. To be checked. 

> For Epec the reasoning is that we recommend the usage of Epic or Epec, so why
> should that be different for Epec_with_sqrt.
@Iordan. Could you please dig into the code of Epec_with_sqrt? Let's discuss offline before answering Andreas here. 
You could also run benchmarks with it against Cartesian<Expr>

MoniqueTeillaud commented:
> @Iordan. Could you please dig into the code of Epec_with_sqrt?
> You could also run benchmarks with it against Cartesian<Expr>

Iordan has checked: Epec_with_sqrt is actually a typedef to Simple_cartesian<Expr> 
and benchmarks show that it is faster than Cartesian<Expr> for our Delaunay triangulations. 
So he will soon push the change of kernel in the corresponding traits (@Iordan please run the testsuite before pushing). 
I guess that there is no change to be done in the manual 
@Andreas thanks for the question

imiordanov commented:
>  I guess that there is no change to be done in the manual @andreas thanks for the question

Perhaps a small change for the default values of the Kernels in the Traits classes..? But I guess it's a rather trivial modification.

MoniqueTeillaud commented:
>> I guess that there is no change to be done in the manual
> Perhaps a small change for the default values of the Kernels in the Traits
> classes..?
I was not sure that this was documented, I did not remember. 
If it is documented then of course ithe manual must be updated consistently with the code.

imiordanov commented:
Thank you @afabri for the comment, indeed the benchmarks are better now, and the code neater!
I will add the new execution times in the User Manuals as soon as I have them (for now I've fixed just the default parameters for the `Kernel` of the traits classes).

afabri commented:
I just adressed the workaround needed for VC2013.
There is still a warning with g++ concerning int/unsigned int. Please fix this.

> include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_2.h:636:34: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Wsign-compare]
           for(unsigned int l=0; l<static_cast<int>(new_f.size()); ++l)

And there is a warning 

>include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_2.h:575:19: warning: variable 'f' set but not used [-Wunused-but-set-variable]
       Face_handle f = e.first;

where the `for` loop around the assignment makes no sense. Either it can be removed or there is a bug.

maxGimeno commented on Hyperbolic_triangulation_2/demo/Hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -0,0 +1,38 @@
+set(CMAKE_AUTOMOC ON) 
+
+cmake_minimum_required(VERSION 2.8.11)
+if(POLICY CMP0043)
```
This should be 
if(NOT POLICY CMP0070 AND POLICY CMP0053)

to silence the warning, because after CMake 3.10 there is a deprecation warning.

lrineau replied:
Actually, there is really no reason now to set `CMP0043`, and the minimal CMake version must be 3.1.

maxGimeno replied:
same for the other demo.

afabri replied:
@maxGimeno  can you please just fix it.

afabri commented:
There is something going wrong in HT2 and PHT2 when GMP is not installed and hence Core not available. 
In HT2/examples it was enough not to include `Expr.h` to fix it.  In PHT2 it is not included in the .cpp but we have compilation errors.
The easy fix is to require Core and to test for CGAL_Core_FOUND in the CMakeLists.txt.

maxGimeno commented:
@imiordanov can you please add me as a collaborator to this repository so I can push some fixes ?

imiordanov commented:
> @imiordanov can you please add me as a collaborator to this repository so I can push some fixes ?

Hello @maxGimeno, I just added you a collaborator. Sorry for the delay...

maxGimeno commented:
Remaining errors : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-101.shtml#Hyperbolic_triangulation_2_Demo

and :
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-101.shtml#Periodic_4_hyperbolic_triangulation_2

afabri commented:
We have several tests failing
* We will have a look for VC++
* @maxGimeno fixes the platforms without GMP
 @imiordanov can you please tell us if you can reproduce the runtime errors and fix them or ask us to fix them if you do not have access to such a compiler version.

imiordanov commented:
> We have several tests failing
> 
> * We will have a look for VC++
> * @maxGimeno fixes the platforms without GMP
>   @imiordanov can you please tell us if you can reproduce the runtime errors and fix them or ask us to fix them if you do not have access to such a compiler version.

Hi @afabri, unfortunately, I am not able to reproduce the errors... 

For HT2, the error reported is: `Unknown CMake command "qt5_wrap_ui"`, while on both my machines (Mac and Ubuntu) Qt wraps the UI files without a problem.

For P4HT2 there is the same `qt5_wrap_ui` error in some cases, plus some errors with the removal test (for which I know the cause and I will push a fix right away). Moreover, in some of the cases the problem is that GMP is not available. For Qt and GMP, I am not sure how I should proceed, I am not able to obtain these errors.

maxGimeno commented:
Remaining warnings in the Examples : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-103.shtml#Periodic_4_hyperbolic_triangulation_2_Examples

We'll try and take care of the errors with MSVC

lrineau commented:
TODO: add `else()` part in the `CMakeLists.txt` of demos, so that the testsuite results are blue when CGAL_Qt5 is not found.

maxGimeno commented:
It is done

lrineau commented:
[**@maxGimeno**](https://github.com/maxGimeno) commented on [Jan 23, 2019, 10:46 AM GMT+1](https://github.com/CGAL/cgal/pull/3283#issuecomment-456736475 "2019-01-23T09:46:17Z - Replied by Github Reply Comments"):
> It is done

That was fast! :-) Thank.

maxGimeno commented:
I actually did it before I saw the message, when you told me IRL.

lrineau commented:
https://travis-ci.org/CGAL/cgal/jobs/483318189#L2738
```
.. Testing permissions...
Hyperbolic_triangulation_2/demo/Hyperbolic_triangulation_2/include/internal/Qt/HyperbolicPainterOstream.h
Periodic_4_hyperbolic_triangulation_2/demo/Periodic_4_hyperbolic_triangulation_2/CMakeLists.txt
Periodic_4_hyperbolic_triangulation_2/demo/Periodic_4_hyperbolic_triangulation_2/include/internal/Qt/HyperbolicPainterOstream.h
Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h
```

Windows is really painful...

maxGimeno commented:
Every single time I write something on windows -_-'

lrineau commented:
TODO: Filter branch, current weight is 4498kB

maxGimeno commented on Hyperbolic_triangulation_2/demo/Hyperbolic_triangulation_2/CMakeLists.txt:
```diff
@@ -19,20 +19,19 @@ find_package(CGAL QUIET COMPONENTS Core Qt5)
 find_package(Qt5 QUIET COMPONENTS Widgets)
 
-find_package( Boost REQUIRED )
+find_package( Boost REQUIRED COMPONENTS thread )
```
I think it is required by CGAL_Core somehow, if it is not here the linker complains.

lrineau replied:
I am really surprised. The user code should not care about that: if `CGAL_Core` requires Boost.thread, then it is registered in the properties of the target `CGAL::CGAL_Core`.

lrineau replied:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-111/Hyperbolic_triangulation_2_Demo/TestReport_lrineau_Ubuntu-Latest.gz
```
CMake Error at CMakeLists.txt:36 (add_executable):
  Target "HDT2_demo" links to target "Boost::thread" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```

That is probably a bug of `CGALConfig.cmake`. Fixing that issue in the user code only hides the bug.

maxGimeno commented:
Remaining problems with Core, apparently : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-113.shtml#Periodic_4_hyperbolic_triangulation_2

afabri commented:
> 
> 
> Remaining problems with Core, apparently : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-113.shtml#Periodic_4_hyperbolic_triangulation_2

This is not a problem of this new package. but a problem with Core.

@lrineau I guess we have again a problem with the destruction order. The memory pool seems to be gone away, before all number objects.

lrineau commented:
# CMake issues
I have fix the CMake issues. There was a bug in `CGALConfig.cmake` when non-header-only.

# Copy of resources (icons, images) from GraphicsView
I see that the demo copies resources from `GraphicsView/demo/icons/*` to `Hyperbolic_triangulation_2/demo/Hyperbolic_triangulation_2/resources/*`. To avoid that, so far we regrouped all 2D demos in `GraphicsView/demo/`. But maybe we should soon or later find another way, with a bit of CMake programming.

Let's say that we keep it that way for CGAL-4.14, but we should do something about our 2D demos.

# Incorrect "About..." dialog

As the resources of the demo were copied blindly from GraphicsView, then the "About..." dialog of the demo is incorrect:
![screenshot_20190204_151512](https://user-images.githubusercontent.com/5746675/52213540-4a138500-2887-11e9-985a-3c5c56dc37d2.png)

afabri commented:
> 
> 
> > Remaining problems with Core, apparently : https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-113.shtml#Periodic_4_hyperbolic_triangulation_2
> 
> This is not a problem of this new package. but a problem with Core.
> 
> @lrineau I guess we have again a problem with the destruction order. The memory pool seems to be gone away, before all number objects.

The problem was not really in Core.  This new package had a static data member, namely a `map<string,FT>` with `FT=Core`.   Making it thread local solves the problem.  @imiordanov

lrineau commented:
On my side, I have fixed the new GMPXX issues (triggered by my CMake fix in https://github.com/CGAL/cgal/pull/3283/commits/24e96c270a5396e0a30e1a501047e8cab990917a), in commits https://github.com/CGAL/cgal/pull/3283/commits/2c179e35fb54ddb09c2df389c93a2c71d159f453 and https://github.com/CGAL/cgal/pull/3283/commits/1c8132ace40e9d09732d05829e136e4f2b91df3d.

afabri commented:
@imiordanov there remain one or two things to do.  Fix the About box and have a look at the [testsuite ](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-116/Periodic_4_hyperbolic_triangulation_2/TestReport_lrineau_CentOS6-CXX11-Boost157.gz) that shows a runtime error.   
@lrineau can you confirm that there is nothing special concerning this testplatform.

lrineau commented:
[**@afabri**](https://github.com/afabri) commented on [Feb 6, 2019, 12:11 PM GMT+1](https://github.com/CGAL/cgal/pull/3283#issuecomment-460986091 "2019-02-06T11:11:18Z - Replied by Github Reply Comments"):
> @imiordanov there remain one or two things to do.  Fix the About box and have a look at the [testsuite ](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-116/Periodic_4_hyperbolic_triangulation_2/TestReport_lrineau_CentOS6-CXX11-Boost157.gz) that shows a runtime error.   
> @lrineau can you confirm that there is nothing special concerning this testplatform.

That depends what you call "special". That platform is using a pre-version of C++11 with an old g++ compiler version (4.4), that was published two years before C++11 got officially voted. That is far from being mainstream.

gcc-4.4 cannot be considered as a valid C++11 compiler (C++11 support is gcc 4.8 and later), and once CGAL enforces C++11 or C++14, we will kick out that test platform, for sure. If you ask, I am even for kicking it out, now, for CGAL-4.14, because that customer of ours has not contacted us for more than three years.

afabri commented:
> If you ask, I am even for kicking it out,

So let me ask

lrineau commented:
I have removed `CentOS6-CXX11-Boost157` from the set of tested Docker images.

maxGimeno commented:
There are runtime errors in two other platforms too.

imiordanov commented:
> @imiordanov there remain one or two things to do. Fix the About box and have a look at the [testsuite ](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-116/Periodic_4_hyperbolic_triangulation_2/TestReport_lrineau_CentOS6-CXX11-Boost157.gz) that shows a runtime error.
> @lrineau can you confirm that there is nothing special concerning this testplatform.

Hello @afabri  and @maxGimeno, thank you for your comments!
Sorry for the late reply, I will try to get on it within the day.

afabri commented:
@imiordanov just in case it is not clear how to reproduce a test that does not fail systematically, you should have a  look at a test that fails, and use the seed you see in:

> CGAL::Random()::get_seed() = 1549527886

You put then into your local test `CGAL::get_default_random() = CGAL::Random(1549527886);`

This won't help if the non-systematic failure is due to non-determinism of the code.

imiordanov commented:
I have fixed the 'About' boxes for both demos; it remains to be verified if the link to the online documentation works (when the documentation is published).

The other problem (runtime error in the testsuite) requires some further investigation, unfortunately I am not able to give a quick solution right away. I know why it happens* and I am able to reproduce it (thank you @afabri, your comment was really helpful!), but I will need to find a good solution because quick fixes don't seem to do the trick.

*For info: when removing vertices, the triangulation must remain a simplicial complex. The error is an assertion failure indicating that the triangulation is _not_ a simplicial complex. There exists a function called `is_removable()` that tests whether a vertex can be removed, but it is based purely on the comparison of the maximum circumdiameter of the faces that would result after the removal of a vertex with the systole of the surface. I will do my best to implement a more sophisticated control during the weekend so that a removal does not cause such issues.

afabri commented:
@imiordanov note that there are still [runtime errors](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-119.shtml)

imiordanov commented:
> @imiordanov note that there are still [runtime errors](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-119.shtml)

The errors should be fixed now, I haven't had an error in over 1000 tries locally from the tests that fail on the testsuite.

afabri commented:
@imiordanov In case this is not done yet, I would like to suggest that you also add test with points on a grid or on a circle or whatever non-general position means in the hyperbolic and periodic hyperbolic case.

maxGimeno commented:
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-121.shtml

I rebased on master to fix a conflict in travis.

lrineau commented:
[**@maxGimeno**](https://github.com/maxGimeno) commented on [Feb 12, 2019, 3:59 PM GMT+1](https://github.com/CGAL/cgal/pull/3283#issuecomment-462791960 "2019-02-12T14:59:16Z - Replied by Github Reply Comments"):
> I rebased on master to fix a conflict in travis.

You did not rebase: you merged `master`. That is a completely different action. `git-rebase` is a scary operation, that rewrite the whole history of a branch, but `git-merge` is just the normal workflow.

afabri commented:
@imiordanov did you see my request for testing degenerate cases?

imiordanov commented:
> @imiordanov did you see my request for testing degenerate cases?

Hi @afabri, yes I did, but unfortunately I didn't have the time to add a test for degenerate cases.
I will try to do it tonight.

maxGimeno commented:
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-124.shtml

lrineau commented:
Now that all the work on the branch is over, @sloriot can you have a look at rewriting the history to remove the big files and reduce the size of the branch?

sloriot commented:
I've cleaned up the branch, the size went from 14Mo to 2.8Mo. I have removed some unused files as well:
```Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/ht-450px.png
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/ht-empty-disks.pdf
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-120px.png
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-250px.png
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-350px.png
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-500px.png
Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/hyperbolic-vs-euclidean.pdf
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/P4T_ds_doc_300.png
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/P4T_dt_300.png
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/data-structure-face-base.pdf
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/new-triangulation-original.png
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/non-triangulation.ipe
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/non-triangulation.png
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/triangulation-120px.png
Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/triangulation-350px.png
```
@imiordanov if you are OK with that change you'll have to reset your local copy of your branch by doing something like:
```
cd my_cgal_checkout
git checkout  Periodic_4_hyperbolic_triangulation_2-IIordanov
git fetch mine
git reset --hard mine/Periodic_4_hyperbolic_triangulation_2-IIordanov
```

The last command is the one that will drop your changes, before running it you can still compare your _old_ branch (which is the local version you have) with the one I filtered (`mine/Periodic_4_hyperbolic_triangulation_2-IIordanov`).

imiordanov commented:
> I've cleaned up the branch, the size went from 14Mo to 2.8Mo. I have removed some unused files as well:
> 
> ```
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/ht-empty-disks.pdf
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-120px.png
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-250px.png
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-350px.png
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/htriangulation-500px.png
> Hyperbolic_triangulation_2/doc/Hyperbolic_triangulation_2/fig/hyperbolic-vs-euclidean.pdf
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/P4T_ds_doc_300.png
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/P4T_dt_300.png
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/data-structure-face-base.pdf
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/new-triangulation-original.png
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/non-triangulation.ipe
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/non-triangulation.png
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/triangulation-120px.png
> Periodic_4_hyperbolic_triangulation_2/doc/Periodic_4_hyperbolic_triangulation_2/fig/triangulation-350px.png
> ```
> @imiordanov if you are OK with that change you'll have to reset your local copy of your branch by doing something like:
> 
> ```
> cd my_cgal_checkout
> git checkout  Periodic_4_hyperbolic_triangulation_2-IIordanov
> git fetch mine
> git reset --hard mine/Periodic_4_hyperbolic_triangulation_2-IIordanov
> ```
> The last command is the one that will drop your changes, before running it you can still compare your _old_ branch (which is the local version you have) with the one I filtered (`mine/Periodic_4_hyperbolic_triangulation_2-IIordanov`).

Thank you @sloriot, I will update my local branch. Thank you for the indications also!

maxGimeno commented:
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-130.shtml

lrineau commented:
This PR is now merged. Congratulation, @imiordanov @MoniqueTeillaud!

@MaelRL, can you please post now the blog entry?
> - [ ] Post this: https://gist.github.com/MaelRL/b161f890b318aa4b05853d0754de7ddf when the PR is integrated

MaelRL commented:
Will do, but it requires the online documentation of master to be updated so it will have to (impatiently) wait until Monday.

imiordanov commented:
> This PR is now merged. Congratulation, @imiordanov @MoniqueTeillaud!
> 
> @MaelRL, can you please post now the blog entry?
> 
> > * [ ]  Post this: https://gist.github.com/MaelRL/b161f890b318aa4b05853d0754de7ddf when the PR is integrated

Excellent news, thank you @lrineau!
Thank you @MaelRL, @afabri, @sloriot, and @maxGimeno for all your comments and contributions!

MoniqueTeillaud commented:
great! 
Many thanks to all

MoniqueTeillaud commented:
Thank you, Mael! 
for the page https://www.cgal.org/2019/02/25/Hyperbolic_triangulations/ 

Details about the line: [ https://www.inria.fr/ | INRIA Nancy Nancy - Grand Est ] , [ https://www.loria.fr/en/ | LORIA ] 
* Nancy appears twice in our affiliation
* the link to INRIA could be replaced by https://www.inria.fr/centre/nancy

Best,
Monique

----- Le 22 Fév 19, à 17:03, Mael <notifications@github.com> a écrit : 

> Will do, but it requires the online documentation of master to be updated so it
> will have to (impatiently) wait until Monday.

MaelRL commented:
Should be fixed now.

MoniqueTeillaud commented:
I should have given the link to the page in English: https://www.inria.fr/en/centre/nancy 
Sorry...
